File: src/main/java/net/valion/manyflowers/ManyFlowers.java
package net.valion.manyflowers;

import me.shedaniel.autoconfig.AutoConfig;
import me.shedaniel.autoconfig.serializer.JanksonConfigSerializer;
import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
import net.valion.manyflowers.config.MFConfig;
import net.valion.manyflowers.helpers.SoundsHelper;
import net.valion.manyflowers.helpers.WorldsHelper;
import net.valion.manyflowers.particle.FlowerParticles;
import net.valion.manyflowers.setup.BlockEntitiesReg;
import net.valion.manyflowers.setup.ItemGroupSetup;
import net.valion.manyflowers.setup.ItemsReg;
import net.valion.manyflowers.world.gen.ModWorldGen;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class ManyFlowers implements ModInitializer {
    public static MFConfig CONFIG;
    public static final String MOD_ID = "many_flowers";
    public static final Logger LOGGER = LogManager.getLogger(MOD_ID);

    @Override
    public void onInitialize() {
        ItemGroupSetup.registerItemGroups();
        ItemsReg.registerModItems();
        FlowerParticles.registerParticles();
        BlockEntitiesReg.register();

        AutoConfig.register(MFConfig.class, JanksonConfigSerializer::new);
        CONFIG = AutoConfig.getConfigHolder(MFConfig.class).getConfig();

        ModWorldGen.generateModWorldGen();

        ServerLifecycleEvents.SERVER_STARTED.register(server -> {
            SoundsHelper.putSounds();
            server.getWorlds().forEach(WorldsHelper::putWorld);
        });
    }
}

File: src/main/kotlin/net/valion/manyflowers/ManyFlowers.kt
package net.valion.manyflowers

import net.fabricmc.api.ModInitializer
import net.fabricmc.fabric.api.event.lifecycle.v1.ServerEntityEvents
import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents
import net.valion.manyflowers.config.MFConfig
import net.valion.manyflowers.event.EntityLoadEvent
import net.valion.manyflowers.event.ServerStartEvent
import net.valion.manyflowers.registry.*
import net.valion.manyflowers.world.gen.ModWorldGeneration
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import kotlin.random.Random


object ManyFlowers: ModInitializer {
    const val MOD_ID: String = "many_flowers"
    val CONFIG: MFConfig = MFConfig.HANDLER.instance()
    val LOGGER: Logger = LogManager.getLogger(MOD_ID)
    val RANDOM: Random = Random(System.currentTimeMillis())

    override fun onInitialize() {
        ItemsRegistry
        BlocksRegistry
        BlocksEntitiesRegistry
        ParticlesRegistry
        FeaturesRegistry
        SoundsRegistry
        EntitiesTypeRegistry
        EntityAttributesRegistry

        ServerEntityEvents.ENTITY_LOAD.register(EntityLoadEvent)
        ServerLifecycleEvents.SERVER_STARTED.register(ServerStartEvent)

        MFConfig.HANDLER.load()
        MFConfig.HANDLER.save()

        ModWorldGeneration.generateModWorldGen()
    }
}